### PR TITLE
Add function to get proxy dicts for Requests

### PR DESCRIPTION
**Pull request by nwithan8**
Adds a `proxy_dict()` function to retrieve a ready-to-use `proxies` parameter for the `requests` library.

### CODE REVIEW

**Comment by nwithan8** [comment]
Unsure if HTTPS needed

File: proxlist/proxies.py [hunk 1]
```diff
@@ -30,3 +30,11 @@ def random_proxy() -> str:
 
 def list_proxies() -> List[str]:
     return PROXY_LIST
+
+
+def proxy_dict() -> dict:
```
Comment (Justintime50) [inline proxlist/proxies.py]:
```suggestion
def proxy_dict() -> Dict[str, str]:
```

Comment (Justintime50) [reply proxlist/proxies.py]: For Python to be happy with type hints on Python prior to 3.9, you'll want to add `from typing import Dict` to the top.

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Thanks for the contribution @nwithan8! This is a nice utility helper. I have some thoughts (I hold my open source to an incredibly high standard, I apologize if some of this is a bit much).

1. Can we add a test function for this under `/test/unit/test_proxies.py`?
2. `https` here is definitely needed, let's keep it. (answering your other question)
3. Nit: I like to assign my return values to a variable so I can return a clean variable. Can we also add a line-break between the return statement and whatever came before it? Makes the code read really nice. Finally, let's maybe add a doctoring here describing what the intended use of this function is as it may not be immediately understood by others. Something such as "Returns a proxies dictionary, perfect for passing into the `requests` proxies parameter" or something to that affect.
4. We'll then want to document this in the README.
5. Finally, we'll want to add this function to the `__init__.py` file imports and include it in the `__all__` variable so users can easily use this once they've imported this package.

**Comment by nwithan8** [comment]
1. [ x]
2. [x]
3. [ x]
4. [ x]
5. [ x]

File: proxlist/proxies.py [hunk 2]
```diff
@@ -29,4 +29,21 @@ def random_proxy() -> str:
...
+    
     return PROXY_LIST
+
+
+def proxy_dict() -> Dict[str, str]:
+    """
+    Returns a proxy dictionary, perfect for passing into the requests `proxies` parameter
+    
+    :return: A dictionary of proxy details
+    :rtype: dict
```
Comment (Justintime50) [inline proxlist/proxies.py:41]: No need for this line since we have type hinting

File: proxlist/proxies.py [hunk 3]
```diff
@@ -29,4 +29,21 @@ def random_proxy() -> str:
 
 
 def list_proxies() -> List[str]:
+    
     return PROXY_LIST
+
+
+def proxy_dict() -> Dict[str, str]:
+    """
+    Returns a proxy dictionary, perfect for passing into the requests `proxies` parameter
```
Comment (Justintime50) [inline proxlist/proxies.py:38]: This should be on the same line as the opening quotes

File: proxlist/proxies.py [hunk 4]
```diff
@@ -29,4 +29,21 @@ def random_proxy() -> str:
...
+    
+    :return: A dictionary of proxy details
+    :rtype: dict
+    """
+    proxy = random_proxy()
+    proxy_dict = {
+        'http': f'http://{proxy}',
+        'https': f'https://{proxy}',
```
Comment (Justintime50) [inline proxlist/proxies.py:46]:
```suggestion
        'https': f'http://{proxy}',
```

File: test/unit/test_proxies.py [hunk 5]
```diff
@@ -11,3 +11,16 @@ def test_list_proxies():
     proxy_list = proxlist.list_proxies()
 
     assert proxy_list == proxlist.proxies.PROXY_LIST
+
+    
+def test_proxy_dict():
+    proxy_dict = proxlist.proxy_dict()
+    http = proxy_dict.get('http', None)
```
Comment (Justintime50) [inline test/unit/test_proxies.py:18]:
```suggestion
    http = proxy_dict.get('http')
```
None is the implicit default here, no need to include

File: test/unit/test_proxies.py [hunk 6]
```diff
@@ -11,3 +11,16 @@ def test_list_proxies():
...
+    https = proxy_dict.get('https', None)
+
+
```
Comment (Justintime50) [inline test/unit/test_proxies.py:21]: No need for two line breaks, one will do

**Review by Justintime50 — CHANGES_REQUESTED** [review]

**Comment by Justintime50** [comment]
Looks like the build also failed. You'll want to run `make black` on the project to format it correctly.

File: test/unit/test_proxies.py (hunk 5 above)
Comment (Justintime50) [reply test/unit/test_proxies.py:18]: Same with the one below it.

**Comment by Justintime50** [comment]
You'll want to rebase this against the `main` branch as I've pushed changes since this PR.

**Comment by Justintime50** [comment]
The `main` branch has diverged so far from this at this point.

I think I'm going to close this as `wontfix` only because I want to keep the library generic, the proxy dictionary although nice is tailored to a `requests` context, the syntax may not always be the same depending on the HTTP library in use.

**Pull request closed without merging**